### PR TITLE
Us22 post bdd

### PR DESCRIPTION
**Pull request by jervalles**
'Backend ready' with express routes
The Front Post from Fil d'actu page will arrive when the actuals PR will merge

### CODE REVIEW

**Review by Gumatan — DISMISSED** [review]

File: backend/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+const bodyParser = require("body-parser");
+app.use(bodyParser.json());
+app.use(
+  bodyParser.urlencoded({
+    extended: true
+  })
+);
+
+app.use(function(req, res, next) {
+  res.header("Access-Control-Allow-Origin", "http://localhost:3001"); // update to match the domain you will make the request from
```
Comment (LoicBrassart) [inline backend/index.js]: Very bad idea: you'll forget to update this value
Instead, use the `cors` package to get rid of this topic in one move :ok_hand:

File: backend/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  res.header(
+    "Access-Control-Allow-Headers",
+    "Origin, X-Requested-With, Content-Type, Accept"
+  );
+  next();
+});
+
+app.get("/", (req, res, next) => {
```
Comment (LoicBrassart) [inline backend/index.js]: Useless route, to delete

File: backend/index.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      if (err) {
+        res.status(500).send("Erreur lors de la récupération des données");
+      } else {
+        res.status(200).json(results);
+      }
+    }
+  );
+});
+
+app.post("/api/users", (req, res) => {
```
Comment (LoicBrassart) [inline backend/index.js]: Beware: this route will be deleted soon once you have nailed the authentication with Passport!

File: backend/index.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      res.status(500).send("Erreur lors de la sauvegarde d'un utilisateur");
+    } else {
+      res.sendStatus(201);
+    }
+  });
+});
+
+// FIL-ACTU || GET & POST
+app.get("/api/posts", (req, res, next) => {
+  db.query("SELECT * from post", (err, results) => {
```
Comment (LoicBrassart) [inline backend/index.js]: Do not "SELECT *" me, young man! è_é

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Some minor changes to do

**Review by Gumatan — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
SnE ;-)

**Pull request merged**